x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_data_grid.tsx

```diff
@@ -0,0 +1,661 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, {
+  ComponentProps,
+  FC,
+  lazy,
+  memo,
+  PropsWithChildren,
+  Suspense,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  EuiCodeBlock,
+  EuiDataGrid,
+  EuiDataGridControlColumn,
+  EuiDataGridProps,
+  EuiDataGridStyle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiSkeletonText,
+  EuiSpacer,
+  EuiText,
+  RenderCellValue,
+} from '@elastic/eui';
+import styled from '@emotion/styled';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { css } from '@emotion/react';
+import { EuiDataGridCellPopoverElementProps } from '@elastic/eui/src/components/datagrid/data_grid_types';
+import { BulkActionsCell } from './bulk_actions/components/row_cell';
+import { BulkActionsHeader } from './bulk_actions/components';
+import { useAlertsTableContext } from './contexts/alerts_table_context';
+import { useBulkActions, useSorting } from './hooks';
+import {
+  AdditionalContext,
+  Alert,
+  AlertsDataGridProps,
+  AlertsTableProps,
+  BulkActionsVerbs,
+  CellActionsOptions,
+  FetchAlertData,
+} from '../../../types';
+import { ALERTS_TABLE_CONTROL_COLUMNS_ACTIONS_LABEL } from './translations';
+
+import './alerts_table.scss';
+import { useGetToolbarVisibility } from './toolbar';
+import { InspectButtonContainer } from './toolbar/components/inspect';
+import { SystemCellId } from './types';
+import { SystemCellFactory, systemCells } from './cells';
+import { typedMemo } from './utils';
+import type { AlertsFlyout as AlertsFlyoutType } from './alerts_flyout/alerts_flyout';
+import { ErrorBoundary } from '../common/components/error_boundary';
+
+const AlertsFlyout = lazy(() => import('./alerts_flyout')) as typeof AlertsFlyoutType;
+
+const DefaultGridStyle: EuiDataGridStyle = {
+  border: 'none',
+  header: 'underline',
+  fontSize: 's',
+};
+
+const defaultCellActionsOptions: CellActionsOptions = {
+  getCellActionsForColumn: () => [],
+  disabledCellActions: [],
+};
+const DEFAULT_PAGE_SIZE_OPTIONS = [10, 20, 50, 100];
+const DEFAULT_ACTIONS_COLUMNS_WIDTH = 75;
+
+const stableMappedRowClasses: EuiDataGridStyle['rowClasses'] = {};
+
+interface BasicCellValueProps {
+  columnId: string;
+  alert: Alert;
+}
+
+const BasicCellValue = memo(({ alert, columnId }: BasicCellValueProps) => {
+  const value = alert[columnId];
+  if (Array.isArray(value)) {
+    return <>{value.length ? value.join() : '--'}</>;
+  }
+  return <>{value}</>;
```

Comment:
It falls back to the second case and an empty cell is rendered. Seems correct?

Reply:
Let's say `value` is `undefined`. The `Array.isArray(value)` will return false and the `return <>{value}</>;` will be returned which will be `undefined`. What do I miss?

Reply:
Yes, and `return <>{undefined}</>` renders nothing in the final DOM, which seems reasonable since the value is undefined. I didn't add a fallback value such as `--` just because this is how other renderers worked, but we can add it.

Reply:
Ok! I was not sure if `return <>{undefined}</>` will return something to the DOM or not.